File: frontend/src/app/core/models/orders-report.models.ts
import { BaseRequest } from './base.models';

/**
 * Sipariş raporu filtreleme ve API isteği modeli
 */
export interface OrdersReportRequest extends BaseRequest {
  /** Başlangıç tarihi (ISO string) */
  startDate: string;
  /** Bitiş tarihi (ISO string) */
  endDate: string;
  /** Cari hesap kodu (Opsiyonel) */
  currentAccountCode?: string;
  /** Sipariş durumu (Opsiyonel) */
  orderStatus?: string;
  /** Ürün fotoğrafı gösterilsin mi? */
  displayProductPhoto: boolean;
  page?: number;
  pageSize?: number;
  sortField?: string;
  sortOrder?: number;
  brand?: string;
}

/**
 * Sipariş raporu liste satırı modeli
 */
export interface OrderReportItem {
  /** Cari hesap kodu */
  currentAccountCode: string;
  /** Cari hesap açıklaması */
  currentAccountDescription: string;
  /** Sipariş detayı/numarası */
  orderDetails: string;
  /** Sipariş durumu */
  status: string;
  /** Marka */
  brand: string;
  /** Sipariş tarihi */
  orderDate: string | Date;
  /** Talep edilen sevkiyat tarihi */
  requestedShipmentDate: string | Date;
  /** Toplam miktar */
  totalQty: number;
  /** İskonto tutarı */
  discount: number;
  /** Toplam tutar */
  total: number;
  /** Sipariş ID (Detaylar için) */
  orderId: number;

  // Edit paneli için eklenen alanlar
  address?: string;
  paymentType?: string;
  shipmentMethod?: string;
  orderDescription?: string;
  subTotal: number;
  grandTotal: number;
  season?: string;
}

/**
 * Sipariş güncelleme isteği
 */
export interface UpdateOrderRequest {
  orderId: number;
  paymentType?: string;
  discountPercent: number;
  requestedShipmentDate?: string;
  shipmentMethod?: string;
  status?: string;
  orderDescription?: string;
  removedProductCodes?: string[];
  warehouseSelections?: OrderWarehouseSelectionDto[];
}

export interface OrderWarehouseSelectionDto {
  productCode: string;
  warehouseId: number;
}

/**
 * Sipariş detay isteği modeli
 */
export interface OrderDetailRequest extends BaseRequest {
  /** Sipariş ID */
  orderId: number;
  /** Marka */
  brand?: string;
  /** Sayfa numarası */
  pageNumber: number;
  /** Sayfa boyutu */
  pageSize: number;
}

/**
 * Sipariş detay yanıt modeli
 */
export interface OrderDetailResponse {
  /** Ürün detayları */
  data: OrderDetailItem[];
  /** Toplam kayıt sayısı */
  totalCount: number;
}

/**
 * Sipariş detay satırı modeli
 */
export interface OrderDetailItem {
  /** Satır ID */
  id: number;
  /** Ürün kodu */
  productCode: string;
  /** Ürün adı */
  productName: string;
  /** Ürün resmi */
  picture?: string;
  /** Miktar */
  qty: number;
  /** Birim fiyat / miktar */
  amount: number;
  /** Satır toplamı */
  total: number;
  /** Satır notu */
  notes?: string;
  /** Ürün özellikleri (JSON) */
  attributes?: string;
}

/**
 * Sipariş kalemi güncelleme isteği
 */
export interface UpdateOrderItemRequest {
  orderId: number;
  productCode: string;
  qty: number;
  amount: number;
  notes?: string;
}

/**
 * Sipariş kalemi silme isteği
 */
export interface RemoveOrderItemRequest {
  orderId: number;
  productCode: string;
}

/**
 * Sipariş raporu yanıt modeli
 */
export interface OrdersReportResponse {
  /** Markalar listesi */
  brands: string[];
  /** Sipariş verileri */
  data: OrderReportItem[];
  totalCount: number;
}

export interface PackingListDto {
  items: PackingListItemDto[];
  cartons: PackingListCartonDto[];
}

export interface PackingListItemDto {
  orderItemId: number;
  productCode: string;
  productName: string;
  qty: number;
  cartonNo: string;
}

export interface PackingListCartonDto {
  id: number;
  cartonNo: string;
  netWeight?: number;
  grossWeight?: number;
  measurements: string;
}

export interface SavePackingListCommand {
  orderId: number;
  items: {
    orderItemId: number;
    cartonNo: string;
    qty: number;
  }[];
  cartons: PackingListCartonDto[];
}

File: backend/src/CWI.Application/DTOs/Reports/OrdersReportModels.cs
using System;
using System.Collections.Generic;

namespace CWI.Application.DTOs.Reports;

/// <summary>
/// Sipariş raporu talep modeli
/// </summary>
public class OrdersReportRequest
{
    public string? CurrentAccountCode { get; set; }
    public string? OrderStatus { get; set; }
    public bool DisplayProductPhoto { get; set; }
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public int Page { get; set; } = 1;
    public int PageSize { get; set; } = 10;
    public string? SortField { get; set; }
    public int SortOrder { get; set; }
    public string? Brand { get; set; }
}

/// <summary>
/// Sipariş raporu kalem modeli
/// </summary>
public class OrderReportDto
{
    public long OrderId { get; set; }
    public string CurrentAccountCode { get; set; } = string.Empty;
    public string CurrentAccountDescription { get; set; } = string.Empty;
    public string OrderDetails { get; set; } = string.Empty;
    public string Status { get; set; } = string.Empty;
    public string Brand { get; set; } = string.Empty;
    public DateTime OrderDate { get; set; }
    public DateTime? RequestedShipmentDate { get; set; }
    public decimal TotalQty { get; set; }
    public decimal Discount { get; set; }
    public decimal Total { get; set; }

    // Edit paneli için eklenen alanlar
    public string? Address { get; set; }
    public string? PaymentType { get; set; }
    public string? ShipmentMethod { get; set; }
    public string? OrderDescription { get; set; }
    public decimal SubTotal { get; set; }
    public decimal GrandTotal { get; set; }
    public string? Season { get; set; }
}

/// <summary>
/// Sipariş güncelleme isteği
/// </summary>
public class UpdateOrderRequest
{
    public long OrderId { get; set; }
    public string? PaymentType { get; set; }
    public decimal DiscountPercent { get; set; }
    public DateTime? RequestedShipmentDate { get; set; }
    public string? ShipmentMethod { get; set; }
    public string? Status { get; set; }
    public string? OrderDescription { get; set; }
    public List<string>? RemovedProductCodes { get; set; }
    public List<OrderWarehouseSelectionDto>? WarehouseSelections { get; set; }
}

public class OrderWarehouseSelectionDto
{
    public string ProductCode { get; set; } = string.Empty;
    public int WarehouseId { get; set; }
}

public class UpdateOrderItemRequest
{
    public long OrderId { get; set; }
    public string ProductCode { get; set; } = string.Empty;
    public int Qty { get; set; }
    public decimal Amount { get; set; }
    public string? Notes { get; set; }
}

public class RemoveOrderItemRequest
{
    public long OrderId { get; set; }
    public string ProductCode { get; set; } = string.Empty;
}

/// <summary>
/// Sipariş detay kalem modeli
/// </summary>
public class OrderDetailDto
{
    public long Id { get; set; }
    public string ProductCode { get; set; } = string.Empty;
    public string ProductName { get; set; } = string.Empty;
    public string? Picture { get; set; }
    public int Qty { get; set; }
    public decimal Amount { get; set; }
    public decimal Total { get; set; }
    public string? Attributes { get; set; }
}

/// <summary>
/// Sipariş detay yanıt modeli
/// </summary>
public class OrderDetailResponse
{
    public List<OrderDetailDto> Data { get; set; } = new();
    public int TotalCount { get; set; }
}

/// <summary>
/// Sipariş raporu yanıt modeli
/// </summary>
public class OrdersReportResponse
{
    public List<string> Brands { get; set; } = new();
    public List<OrderReportDto> Data { get; set; } = new();
    public int TotalCount { get; set; }
}
